flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1367,7 +1367,7 @@ class CodeGenerator(
         val right = operands(1)
         requireTemporal(left)
         requireTemporal(right)
-        generateTemporalPlusMinus(plus = false, nullCheck, left, right)
+        generateTemporalPlusMinus(plus = false, nullCheck, call.`type`.getSqlTypeName, left, right)
```

Comment:
I think you don't need to change this method. You can do the match with the `TimeIntervalTypeInfo`.